src/snapred/ui/view/NormalizationTweakPeakView.py

```diff
@@ -179,6 +179,31 @@ def _updateGraphs(self, peaks):
         self.figure.clear()
         for i in range(numGraphs):
             ax = self.figure.add_subplot(nrows, ncols, i + 1, projection="mantid")
+
+            # NOTE: Mutate the ax object as the mantidaxis does not account for lines
+            # TODO: Bubble this up to the mantid codebase and remove this mutation.
+            def rename_workspace(old_name, new_name):
+                """
+                Rename a workspace, and update the artists, creation arguments and tracked workspaces accordingly
+                :param new_name : the new name of workspace
+                :param old_name : the old name of workspace
+                """
+                for cargs in ax.creation_args:
+                    # NEW CHECK
+                    func_name = cargs["function"]
+                    if func_name not in ["axhline", "axvline"] and cargs["workspaces"] == old_name:
+                        cargs["workspaces"] = new_name
+                    # Alternatively,
+                    # if cargs.get("workspaces") == old_name:
+                    #     cargs["workspaces"] = new_name
+                for ws_name, ws_artist_list in list(ax.tracked_workspaces.items()):
+                    for ws_artist in ws_artist_list:
+                        if ws_artist.workspace_name == old_name:
+                            ws_artist.rename_data(new_name)
+                    if ws_name == old_name:
+                        ax.tracked_workspaces[new_name] = ax.tracked_workspaces.pop(old_name)
+
+            ax.rename_workspace = rename_workspace
```

Comment:
Let's centralise this \<code section\> somewhere, rather than just duplicating it?  I suggest something like `src/snapred/ui/view/mantid_patches` ... or something like that.  AND, we definitely need to write up the mantid defect.